python/src/energuide/extracted_datatypes.py

```diff
@@ -496,3 +583,215 @@ def to_dict(self) -> typing.Dict[str, typing.Union[str, float]]:
+            WaterHeaterType.PROPANE_INDUCED_DRAFT_FAN_PILOT_FRENCH.value,
+        ),
+        ("Propane", "Direct vent (sealed)"): (
+            WaterHeaterType.PROPANE_DIRECT_VENT_SEALED_ENGLISH.value,
+            WaterHeaterType.PROPANE_DIRECT_VENT_SEALED_FRENCH.value,
+        ),
+        ("Propane", "Direct vent (sealed, pilot)"): (
+            WaterHeaterType.PROPANE_DIRECT_VENT_SEALED_PILOT_ENGLISH.value,
+            WaterHeaterType.PROPANE_DIRECT_VENT_SEALED_PILOT_FRENCH.value,
+        ),
+        ("Propane", "Condensing"): (
+            WaterHeaterType.PROPANE_CONDENSING_ENGLISH.value,
+            WaterHeaterType.PROPANE_CONDENSING_FRENCH.value,
+        ),
+        ("Wood Space Heating (Mixed Wood, Hardwood, Soft Wood or Wood Pellets)", "Not applicable"): (
+            WaterHeaterType.NOT_APPLICABLE.value,
+            WaterHeaterType.NOT_APPLICABLE.value,
+        ),
+        ("Wood Space Heating (Mixed Wood, Hardwood, Soft Wood or Wood Pellets)", "Fireplace"): (
+            WaterHeaterType.WOOD_SPACE_HEATING_FIREPLACE_ENGLISH.value,
+            WaterHeaterType.WOOD_SPACE_HEATING_FIREPLACE_FRENCH.value,
+        ),
+        ("Wood Space Heating (Mixed Wood, Hardwood, Soft Wood or Wood Pellets)", "Wood stove water coil"): (
+            WaterHeaterType.WOOD_SPACE_HEATING_WOOD_STOVE_WATER_COIL_ENGLISH.value,
+            WaterHeaterType.WOOD_SPACE_HEATING_WOOD_STOVE_WATER_COIL_FRENCH.value,
+        ),
+        ("Wood Space Heating (Mixed Wood, Hardwood, Soft Wood or Wood Pellets)", "Indoor wood boiler"): (
+            WaterHeaterType.WOOD_SPACE_HEATING_INDOOR_WOOD_BOILER_ENGLISH.value,
+            WaterHeaterType.WOOD_SPACE_HEATING_INDOOR_WOOD_BOILER_FRENCH.value,
+        ),
+        ("Wood Space Heating (Mixed Wood, Hardwood, Soft Wood or Wood Pellets)", "Outdoor wood boiler"): (
+            WaterHeaterType.WOOD_SPACE_HEATING_OUTDOOR_WOOD_BOILER_ENGLISH.value,
+            WaterHeaterType.WOOD_SPACE_HEATING_OUTDOOR_WOOD_BOILER_FRENCH.value,
+        ),
+        ("Wood Space Heating (Mixed Wood, Hardwood, Soft Wood or Wood Pellets)", "Wood hot water tank"): (
+            WaterHeaterType.WOOD_SPACE_HEATING_WOOD_HOT_WATER_TANK_ENGLISH.value,
+            WaterHeaterType.WOOD_SPACE_HEATING_WOOD_HOT_WATER_TANK_FRENCH.value,
+        ),
+        ("Solar", "Solar Collector System"): (
+            WaterHeaterType.SOLAR_COLLECTOR_SYSTEM_ENGLISH.value,
+            WaterHeaterType.SOLAR_COLLECTOR_SYSTEM_FRENCH.value,
+        ),
+        ("CSA P9-11 tested Combo Heat/DHW", "CSA P9-11 tested Combo Heat/DHW"): (
+            WaterHeaterType.CSA_DHW_ENGLISH.value,
+            WaterHeaterType.CSA_DHW_FRENCH.value,
+        ),
+    }
+
+    @classmethod
+    def _from_data(cls, water_heating: element.Element) -> 'WaterHeating':
+        assert water_heating.attrib['hasDrainWaterHeatRecovery'] == 'false'
+
+        energy_type = water_heating.xpath('EnergySource/English/text()')[0]
```

Comment:
`water_heating.get_text('EnergySource/English')` is a better way to fail (assertion error on an explicit check of non-null value). This will give you an `index out of range` error if `EnergySource/English` doesn't exist for some reason.